src/14-curated-discovery/06-collections.ts

```diff
@@ -0,0 +1,224 @@
+      {
+        name: "short_description",
+        dataType: ["text"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: true,
+          },
+        },
+      },
+      {
+        name: "category",
+        dataType: ["text[]"],
+        description:
+          "The 'name's of the genes pulled from associated 'gene_ids' in Gravity's MarketingCollection Model",
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: false,
+          },
+        },
+      },
+      {
+        name: "group",
+        dataType: ["text"],
+        description:
+          "Renamed key for the value of 'category' in Gravity's MarketingCollection Model",
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: false,
+          },
+        },
+      },
+      {
+        name: "price_guidance",
+        dataType: ["text"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: true,
+          },
+        },
+      },
+      {
+        name: "image_url",
+        dataType: ["text"],
+        moduleConfig: {
+          "text2vec-openai": {
+            skip: true,
+          },
+        },
+      },
+    ],
+  }
+
+  const classResult = await client.schema
+    .classCreator()
+    .withClass(classWithProps)
+    .do()
+
+  console.log(JSON.stringify(classResult, null, 2))
+}
+
+/**
+ * Insert collections into Weaviate
+ *
+ * Also assigns a deterministic UUID to each collection,
+ * based on its Gravity ID
+ */
+async function insertCollections(
+  collections: GravityArtworkCollection[],
+  batchSize: number = BATCH_SIZE
+) {
+  console.log(`Inserting collection: ${collections.length}`)
+
+  const batches = _.chunk(collections, batchSize)
+  console.log(`Inserting ${batches.length} batches`)
+
+  for (const collectionsBatch of batches) {
+    const shortDescriptions = await batchShortDescriptions(collectionsBatch)
+
+    let batcher = client.batch.objectsBatcher()
+    batcher = batcher.withObjects(
+      ...collectionsBatch.map((collection) => {
+        return {
+          class: CLASS_NAME,
+          properties: {
+            internalID: collection.id,
+            category: collection.category,
+            description: collection.description,
+            short_description: shortDescriptions[collection.id],
+            group: collection.group,
+            image_url: collection.image_url,
+            price_guidance: collection.price_guidance,
+            slug: collection.slug,
+            title: collection.title,
+          },
+          id: generateUuid5(collection.id),
+        }
+      })
+    )
+    process.stdout.write(".")
+    await batcher.do()
+  }
+  process.stdout.write("\n")
+}
+
+/**
+ * Return a short description generated by openai using a collection's long description.
+ *
+ * @param description - an artwork collection's long description
+ * @returns a short, one sentence, description
+ */
+async function getShortDescription(description: string) {
```

Comment:
Since we need to utilize the `openai` client I opted to keep this and `batchShortDescriptions` in this file, rather than moving them to `helpers.ts` and passing the client around.